### ISSUE TITLE
reload fails on adding a new xtrigger

### ISSUE THREAD

**Issue by hjoliver** [cylc/cylc-flow#6260]
```ini
[scheduling]
    initial cycle point = now
    [[graph]]
        PT1M = "foo"
[runtime]
    [[foo]]
        script = sleep 10
```
Run the above, change the source to this then `cylc vr` (reload):
```ini
[scheduling]
    initial cycle point = now
    [[xtriggers]]
        x = wall_clock(offset=P1Y)
    [[graph]]
        PT1M = "@x => foo"
[runtime]
    [[foo]]
        script = sleep 10
```

Result:
```console
    Traceback (most recent call last):
      File "/home/oliverh/cylc/cylc-flow/cylc/flow/scheduler.py", line 652, in run_scheduler
        await self._main_loop()
      File "/home/oliverh/cylc/cylc-flow/cylc/flow/scheduler.py", line 1537, in _main_loop
        self.process_queued_task_messages()
      File "/home/oliverh/cylc/cylc-flow/cylc/flow/scheduler.py", line 907, in process_queued_task_messages
        if self.task_events_mgr.process_message(
      File "/home/oliverh/cylc/cylc-flow/cylc/flow/task_events_mgr.py", line 731, in process_message
        self.spawn_children(itask, TASK_OUTPUT_SUCCEEDED)
      File "/home/oliverh/cylc/cylc-flow/cylc/flow/task_events_mgr.py", line 1970, in spawn_children
        self.spawn_func(itask, output)
      File "/home/oliverh/cylc/cylc-flow/cylc/flow/task_pool.py", line 1440, in spawn_on_output
        self.remove_if_complete(itask, output)
      File "/home/oliverh/cylc/cylc-flow/cylc/flow/task_pool.py", line 1496, in remove_if_complete
        self.release_runahead_tasks()
      File "/home/oliverh/cylc/cylc-flow/cylc/flow/task_pool.py", line 276, in release_runahead_tasks
        self.spawn_to_rh_limit(
      File "/home/oliverh/cylc/cylc-flow/cylc/flow/task_pool.py", line 803, in spawn_to_rh_limit
        self.spawn_if_parentless(tdef, point, flow_nums)
      File "/home/oliverh/cylc/cylc-flow/cylc/flow/task_pool.py", line 812, in spawn_if_parentless
        self.add_to_pool(ntask)
      File "/home/oliverh/cylc/cylc-flow/cylc/flow/task_pool.py", line 228, in add_to_pool
        self.create_data_store_elements(itask)
      File "/home/oliverh/cylc/cylc-flow/cylc/flow/task_pool.py", line 239, in create_data_store_elements
        self.data_store_mgr.increment_graph_window(
      File "/home/oliverh/cylc/cylc-flow/cylc/flow/data_store_mgr.py", line 791, in increment_graph_window
        self.generate_ghost_task(
      File "/home/oliverh/cylc/cylc-flow/cylc/flow/data_store_mgr.py", line 1266, in generate_ghost_task
        self._process_internal_task_proxy(itask, tproxy)
      File "/home/oliverh/cylc/cylc-flow/cylc/flow/data_store_mgr.py", line 1521, in _process_internal_task_proxy
        sig = self.schd.xtrigger_mgr.get_xtrig_ctx(
      File "/home/oliverh/cylc/cylc-flow/cylc/flow/xtrigger_mgr.py", line 599, in get_xtrig_ctx
        ctx = deepcopy(self.xtriggers.functx_map[label])
    KeyError: 'x'
CRITICAL - Workflow shutting down - 'x'
```

**Comment by ColemanTom** [comment]
We just hit this, so glad to see its been reported already. Our workaround was to 

1. reload
2. let it disappear
3. `cylc play <workflow name>`

**Comment by hjoliver** [comment]
Hi @ColemanTom - this has already been fixed, and released in 8.3.6

Looks like we forgot to close the issue (merges to side-branches, i.e. 8.3.x as opposed to master) required manual issue closure until a very recent update to GitHub UI.

**Comment by hjoliver** [comment]
Closed by #6263
